### PR TITLE
Added the session one exercises

### PR DESCRIPTION
**Pull request by mlara-bxw**

### CODE REVIEW

File: manuel_lara/session-01/exercise-01/secret.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mariadb-credentials
+  namespace: exercise-01
+  labels:
+    app: exercise-01
+type: Opaque
+stringData:
```
Comment (juan131) [inline manuel_lara/session-01/exercise-01/secret.yml]: Why not using 'encoded' data?

File: manuel_lara/session-01/exercise-01/service.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress
+  namespace: exercise-01
+  labels:
+    app: exercise-01
+spec:
+  type: NodePort
```
Comment (juan131) [inline manuel_lara/session-01/exercise-01/service.yml]: You can consider the cluster has the ability to create `LoadBalancer` SVCs

File: manuel_lara/session-01/exercise-01/service.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress
+  namespace: exercise-01
+  labels:
+    app: exercise-01
+spec:
```
Comment (juan131) [inline manuel_lara/session-01/exercise-01/service.yml]: I'm missing the session affinity solution

File: manuel_lara/session-01/exercise-02/command.bash [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        kubectl delete cm $name --namespace=exercise-02
+    fi
+
+    kubectl create cm $name --from-file=$path --namespace=exercise-02
+}
+
+echo "Cretaing an isolate namespace..."
+kubectl apply -f $HOME/namespace.yml
+
+echo "Setting up some configuration values"
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/command.bash]: Nice solution to include **hyperdb** as a configmap!

File: manuel_lara/session-01/exercise-02/horizontalPodAutoscaler.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/horizontalPodAutoscaler.yml]: Really nice!!! It was not requested but it's obviously a nice solution for auto-scaling!

File: manuel_lara/session-01/exercise-02/deployment.yml [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+              valueFrom:
+                configMapKeyRef:
+                  name: wordpress-cm
+                  key: allow_empty_password
+          ports:
+            - name: http
+              containerPort: 80
+            - name: https
+              containerPort: 443
+          resources:
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/deployment.yml]: Good resources requests/limits!

File: manuel_lara/session-01/exercise-02/deployment.yml [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  template:
+    metadata:
+      labels:
+        name: wordpress
+        app: exercise-02
+        tier: frontend
+    spec:
+      containers:
+        - name: wordpress
+          image: bitnami/wordpress:4.9.8
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/deployment.yml]: Use latest

File: manuel_lara/session-01/exercise-02/deployment.yml [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+                  key: mariadb_database
+            - name: ALLOW_EMPTY_PASSWORD
+              valueFrom:
+                configMapKeyRef:
+                  name: mariadb-cm
+                  key: allow_empty_password
+          ports:
+            - name: tcp
+              containerPort: 3306
+          resources:
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/deployment.yml]: Good resources requests/limits!

File: manuel_lara/session-01/exercise-02/deployment.yml [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    tier: backend
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: mariadb
+        app: exercise-02
+        tier: backend
+    spec:
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/deployment.yml]: I'm missing the **rollingUpdate** strategy definition

File: manuel_lara/session-01/exercise-02/deployment.yml [hunk 10]
```diff
@@ -0,0 +1,150 @@
+apiVersion: extensions/v1beta1
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/deployment.yml]: This is not a master-slave solution!!!

File: manuel_lara/session-01/exercise-02/deployment.yml [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+              port: 80
+            initialDelaySeconds: 10
+            periodSeconds: 10
+          livenessProbe:
+            httpGet:
+              path: /
+              port: 80
+            initialDelaySeconds: 20
+            periodSeconds: 15
+      # volumes:
```
Comment (juan131) [inline manuel_lara/session-01/exercise-02/deployment.yml]: Why is this commented?

**Comment by mlara-bxw** [comment]
Hi Juan, thank you so much for the comments. As you point, there some typos and effectively the master/slave deployment is undone. I have no time yet for that but I going to try to have it for the next session.

Again, thank you so much for the comments.

**Pull request merged**